framework/src/meshgenerators/ParsedCurveGenerator.C

```diff
@@ -216,27 +223,36 @@ ParsedCurveGenerator::generate()
                    "this parameter is not needed if the first and last points of the curve to be "
                    "generated are overlapped.");
     }
-    else if (_forced_closing_num_segments > 1)
+    else if (_forced_closing_num_segments * _order > 1)
     {
       // Add extra nodes on the curve section used to forcefully close the loop
       const Point start_pt(*nodes.back());
       const Point end_pt(*nodes.front());
       const unsigned int num_nodes_0(nodes.size());
-      for (unsigned int i = 1; i < _forced_closing_num_segments; i++)
+      for (unsigned int i = 1; i < _forced_closing_num_segments * _order; i++)
       {
-        Point side_p =
-            start_pt + (end_pt - start_pt) * (Real)i / (Real)_forced_closing_num_segments;
+        Point side_p = start_pt + (end_pt - start_pt) * (Real)i /
+                                      (Real)(_forced_closing_num_segments * _order);
         nodes.push_back(mesh->add_point(side_p, num_nodes_0 + i - 1));
       }
     }
   }
 
-  for (unsigned int i = 0; i < nodes.size() - !_is_closed_loop; i++)
+  const unsigned int num_elems = (nodes.size() - !_is_closed_loop) / _order;
+  for (unsigned int i = 0; i < num_elems; i++)
   {
-    Elem * elem = mesh->add_elem(new Edge2);
-    elem->set_node(0) = nodes[i];
-    elem->set_node(1) = nodes[(i + 1) % nodes.size()];
-    elem->subdomain_id() = 1;
+    std::unique_ptr<Elem> new_elem;
+    new_elem = std::make_unique<Edge2>();
+    if (_order == 2)
+    {
+      new_elem = std::make_unique<Edge3>();
+      new_elem->set_node(2) = nodes[i * _order + 1];
+    }
+    new_elem->set_node(0) = nodes[i * _order];
+    new_elem->set_node(1) = nodes[((i + 1) * _order) % nodes.size()];
+
+    new_elem->subdomain_id() = 1;
+    mesh->add_elem(std::move(new_elem));
```

Comment:
In addition to `EDGE2` and `EDGE3` elements, libMesh also offers `EDGE4` elements. To support `EDGE4`, wouldn't we just need to set `_order` to 3 and set the appropriate nodes here?

As an aside, libMesh also has an `InfEdge2` element, but I'm not sure that infinite elements are applicable here.

Reply:
Added support for EDGE4